# Shell commands to query vcf files and format into desired state.
# Resulting files are tab-delimited text files with fields Chrom,Pos,
# Ref,Alt,Genotype, with genotype encoded as something like 0|1.  See
# .vcf file format description for details.
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr1.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr1.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr2.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr2.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr3.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr3.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr4.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr4.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr5.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr5.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr6.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr6.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr7.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr7.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr8.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr8.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr9.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr9.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr10.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr10.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr11.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr11.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr12.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr12.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr13.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr13.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr14.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr14.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr15.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr15.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr16.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr16.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr17.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr17.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr18.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr18.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr19.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr19.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr20.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr20.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr21.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr21.txt
bcftools query -S EUR_samples.txt -i 'MAF>0.01 & TYPE="snp" & N_ALT=1 & ID=@target_snp_ids.txt' -f "%CHROM\t%POS\t%REF\t%ALT\t%ID[\t%GT]\n" GEUVADIS.chr22.PH1PH2_465.IMPFRQFILT_BIALLELIC_PH.annotv2.genotypes.vcf.gz > GEUVADIS_EUR_hapmapCEU_MAF01_chr22.txt
